java-frontend/src/main/java/org/sonar/java/se/MethodBehavior.java

```diff
@@ -20,48 +20,50 @@
 package org.sonar.java.se;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 
 import org.sonar.java.resolve.JavaSymbol;
+import org.sonar.java.se.constraint.Constraint;
 import org.sonar.java.se.symbolicvalues.SymbolicValue;
 import org.sonar.plugins.java.api.semantic.Symbol;
-import org.sonar.plugins.java.api.tree.VariableTree;
 
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.Set;
 
 public class MethodBehavior {
   private final Symbol.MethodSymbol methodSymbol;
-  private final List<MethodYield> yields;
-  private final Map<Symbol, SymbolicValue> parameters;
+  private final Set<MethodYield> yields;
+  private final List<SymbolicValue> parameters;
 
   public MethodBehavior(Symbol.MethodSymbol methodSymbol) {
     this.methodSymbol = methodSymbol;
-    this.yields = new ArrayList<>();
-    this.parameters = new LinkedHashMap<>();
+    this.yields = new LinkedHashSet<>();
+    this.parameters = new ArrayList<>();
   }
 
-  public void createYield(ProgramState programState) {
-    List<MethodYield.ConstrainedSymbolicValue> parametersCSV = methodSymbol.declaration().parameters().stream()
-      .map(VariableTree::symbol)
-      .map(parameters::get)
-      .map(sv -> new MethodYield.ConstrainedSymbolicValue(sv, programState.getConstraint(sv)))
-      .collect(Collectors.toList());
+  public void addYield(ProgramState programState) {
+
+    MethodYield yield = programState.yield();
+
+    parameters.forEach(sv -> {
+      Constraint constraint = programState.getConstraint(sv);
+      if (constraint != null) {
+        yield.parametersConstraints[parameters.indexOf(sv)] = constraint;
+      }
+    });
 
-    MethodYield.ConstrainedSymbolicValue returnCSV = null;
     if (!isConstructor() && !isVoidMethod()) {
       SymbolicValue returnSV = programState.peekValue();
+      // FIXME Handle exception path
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1134)